repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -311,14 +364,82 @@ class Interpreter(prompt0: Ref[String],
       (classFiles, newImports) <- compiled
       _ = Timer("cachedCompileBlock 4")
       cls <- eval.loadClass(fullyQualifiedName, classFiles)
-    } yield (cls, newImports)
+    } yield (cls, newImports, tag)
+
   }
 
-  def processModule(source: ImportHook.Source, code: String, wrapperName: Name, pkgName: Seq[Name]) = {
 
-    processModule0(source, code, wrapperName, pkgName, predefImports)
+  def evalCachedClassFiles(cachedData: Seq[CompileCache],
+                           pkg: String,
+                           wrapper: String): Res[Unit] = {
+
+    var blockNumber = 1;
+    def getBlockNumber = blockNumber match {
+      case 1 => ""
+      case _ => "_" + blockNumber.toString
+    }
+
+    // Iterate through cached blocks and execute classFiles
+    // blockNumber keeps track of blockIndex
+
+    Res[Unit](Try {
+      cachedData.foreach{ d =>
+         for {
+          cls <- eval.loadClass(pkg + "." + wrapper + getBlockNumber, d._1)
+        } yield {
+           addToClasspath(d._1)
+          eval.evalMain(cls)
+        }
+        blockNumber += 1
+      }
+    }, e => "Failed to Evaluate Cached Class Files " + e)
+
+  }
+
+
+  def processModule(source: ImportHook.Source,
+                    code: String,
+                    wrapperName: Name,
+                    pkgName: Seq[Name]
+                   ): Res[Imports] = {
+    val cacheTag = "cache" + Util.md5Hash(Iterator(code.getBytes)).map("%02x".format(_)).mkString
+    storage.classFilesListLoad(
+      pkgName.map(_.backticked).mkString("."),
+      wrapperName.backticked,
+      cacheTag) match {
+      case (Seq(), Imports(Nil)) =>
+        init()
+        val res = processModule0(source, code, wrapperName, pkgName, predefImports)
+        res match{
+         case Res.Success(data) =>
+           updateClassPath
+           val Imports(imports) = data._1
+           val files = for (d <- data._2) yield (d._1, d._2)
+           if (imports.count(_.toName == "main") == 0)
+             storage.classFilesListSave(pkgName.map(_.backticked).mkString("."),
+               wrapperName.backticked,
+               files,
+               data._1,
+               cacheTag)
+           res.map(_._1)
+         case r =>
+           r.asInstanceOf[Res[Imports]]
+       }
+      case (cachedData, imports) =>
+        evalCachedClassFiles(cachedData,
+          pkgName.map(_.backticked).mkString("."),
+          wrapperName.backticked) match {
+          case Res.Success(_) =>
+            Res.Success(imports)
+          case r => r.asInstanceOf[Res[Imports]]
+
+        }
+    }
   }
 
+
```

Comment:
Why are you adding three extra blank lines here? It has nothing to do with your change, and is more lines than we should need anyway: 1 or 2, which we already had before